test/mock.coffee

```diff
@@ -25,12 +26,47 @@ describe "require('clever/mock') [API KEY] [MOCK DATA DIR]", ->
       assert.deepEqual _(data).invoke('toJSON'), require("#{__dirname}/mock_data/students")
       done()
 
+  it "deep copies data", (done) ->
+    async.waterfall [
+      (cb_wf) =>
+        @clever.Student.find().exec (err, students) ->
+          assert.ifError err
+          name = students[0].get 'name'
+          assert.equal name.first, "John"
+          name.first = 'WRONG NAME'
+          cb_wf()
+      (cb_wf) =>
+        @clever.Student.find().exec (err, students) ->
+          assert.ifError err
+          name = students[0].get 'name'
+          assert.equal name.first, "John"
+          cb_wf()
+    ], done
+
+
   it "supports GETting properties", (done) ->
     @clever.Student.find().exec (err, students) => # TODO: get findOne working
       students[0].properties (err, data) =>
         assert.ifError err
         assert.deepEqual data, _(require("#{__dirname}/mock_data/studentproperties")).findWhere({student: students[0].get('id')}).data
-      done()
+        done()
+
+  it "supports deep copies of properties", (done) -> # depends on previous test
+    async.waterfall [
+      (cb_wf) =>
+        @clever.Student.find().exec (err, students) =>
+          students[0].properties (err, data) =>
+            assert.ifError err
+            assert.equal data.foo, "bar"
+            data.foo = 'WRONG FOO'
+            cb_wf()
+      (cb_wf) =>
+        @clever.Student.find().exec (err, students) =>
+          students[0].properties (err, data) =>
+            assert.ifError err
+            assert.equal data.foo, "bar"
+            cb_wf()
+    ], done
```

Comment:
Can these tests take advantage of the [inequality operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators) and corresponding [assert](http://nodejs.org/api/assert.html#assert_assert_notequal_actual_expected_message) function? Specifically:

>If both operands are objects, then JavaScript compares internal references which are not equal when operands refer to different objects in memory.

So something like
```
get_properties = (cb) ->
@clever.Student.find().exec (err, students) =>
  students[0].properties cb
properties1 = properties2 = null
async.auto [
  props1: (cb_a) ->
    get_properties cb_a
  props2: (cb_a) ->
    get_properties cb_a
], (err, results) ->
  assert.notEqual results.props1, results.props2, "props1 and props2 are the same object in memory"
```